ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallResult.kt

```diff
@@ -0,0 +1,30 @@
+package com.revenuecat.purchases.ui.revenuecatui.activity
+
+import android.os.Parcelable
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.PurchasesError
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Result of the paywall activity.
+ */
+sealed class PaywallResult : Parcelable {
```

Comment:
Is this usable from Java too?

Reply:
It's usable yes, though you don't really have a nice switch as you do in Kotlin, you basically have to check the type and perform the downcast to the corresponding type. It's really a much nicer API in Kotlin, but still usable in Java.

Reply:
👍🏻